File: src/main/java/org/radarbase/appserver/exception/InvalidPathDetailsException.java
package org.radarbase.appserver.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.EXPECTATION_FAILED)
public class InvalidPathDetailsException extends IllegalArgumentException {
    private static final long serialVersionUID = -793674245766939L;

    public InvalidPathDetailsException(String message) {
        super(message);
    }

    public InvalidPathDetailsException(String message, Object object) {
        super(message + " " + object.toString());
    }
}

File: appserver-legacy/src/main/kotlin/org/radarbase/appserver/exception/InvalidPathDetailsException.kt
package org.radarbase.appserver.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import java.io.Serial

@Suppress("unused")
@ResponseStatus(HttpStatus.EXPECTATION_FAILED)
class InvalidPathDetailsException : IllegalArgumentException {
    companion object {
        @Serial
        private const val serialVersionUID = -793674245766939L
    }

    constructor(message: String) : super(message)

    constructor(message: String, obj: Any) : super("$message $obj")
}
